Lib/ufo2ft/instructionCompiler.py

```diff
@@ -0,0 +1,194 @@
+                    glyf.program.fromBytecode(glyf.program.getBytecode())
+
+            # Handle composites
+            if glyf.isComposite():
+                # Remove empty glyph programs from composite glyphs
+                if hasattr(glyf, "program") and not glyf.program:
+                    delattr(glyf, "program")
+
+                # Set component flags
+
+                # We need to decide when to set the flags.
+                # Let's assume if any lib key is not there, or the component
+                # doesn't have an identifier, we should leave the flags alone.
+                use_my_metrics_comp = None
+                for i, c in enumerate(glyf.components):
+                    if i >= len(glyph.components):
+                        logger.error(
+                            "Number of components differ between UFO and TTF "
+                            f"in glyph '{name}' ({len(glyph.components)} vs. "
+                            f"{len(glyf.components)}, not setting flags in "
+                            "additional components."
+                        )
+                        break
+                    ufo_component_id = glyph.components[i].identifier
+                    if (
+                        ufo_component_id is not None
+                        and OBJECT_LIBS_KEY in glyph.lib
+                        and ufo_component_id in glyph.lib[OBJECT_LIBS_KEY]
+                        and (
+                            TRUETYPE_ROUND_KEY
+                            in glyph.lib[OBJECT_LIBS_KEY][ufo_component_id]
+                            or TRUETYPE_METRICS_KEY
+                            in glyph.lib[OBJECT_LIBS_KEY][ufo_component_id]
+                        )
+                    ):
+                        component_lib = glyph.lib[OBJECT_LIBS_KEY][ufo_component_id]
+
+                        c.flags &= ~ROUND_XY_TO_GRID
+                        if component_lib.get(TRUETYPE_ROUND_KEY, False):
+                            c.flags |= ROUND_XY_TO_GRID
+
+                        c.flags &= ~USE_MY_METRICS
+                        if component_lib.get(TRUETYPE_METRICS_KEY, False):
+                            if use_my_metrics_comp:
+                                logger.warning(
+                                    "Ignoring USE_MY_METRICS flag on component "
+                                    f"'{ufo_component_id}' because it has been set on "
+                                    f"component '{use_my_metrics_comp}' already."
+                                )
+                            else:
+                                c.flags |= USE_MY_METRICS
+                                use_my_metrics_comp = ufo_component_id
+
+                    # We might automatically set the flags if no data is present,
+                    # but:
+                    # - https://github.com/googlefonts/ufo2ft/pull/425 recommends
+                    #   against setting the ROUND_XY_TO_GRID flag
```

Comment:
"against"? I re-read that issue and it seems like the consensus was to always set ROUND_XY_TO_GRID.

> for modern rendering environments, the flag doesn't seem to matter; for ClearType it is better to always set it.

Reply:
Oops ... I probably wrote this before that discussion was finished. I changed the default behaviour to always set ROUND_XY_TO_GRID, and only unset it if explicitly set to false in the lib. (in d8362207d1ff8a830fe72651101f1c3abfc9ad25)